nbgrader/apps/baseapp.py

```diff
@@ -52,26 +71,90 @@ class BaseNbGraderApp(BaseIPythonApplication):
 
     """
 
-    aliases = Dict(nbgrader_aliases)
-    flags = Dict(nbgrader_flags)
+    aliases = Dict(base_aliases)
+    flags = Dict(base_flags)
 
-    profile = Unicode('nbgrader', config=True, help="Default IPython profile to use")
-    auto_create = Bool(True, config=True, help="Whether to automatically generate the profile")
+    verbose_crash = Bool(False)
+
+    # This is a hack in order to centralize the config options inherited from
+    # IPython. Rather than allowing them to be configured for each app separately,
+    # this makes them non-configurable for the apps themselves. Then, in the init,
+    # an instance of `BasicConfig` is created, which contains these options, and
+    # their values are set on the application instance. So, to configure these
+    # options, the `BasicConfig` class can be configured, rather than the
+    # application itself.
+    profile = Unicode()
+    overwrite = Bool()
+    auto_create = Bool()
+    extra_config_file = Unicode()
+    copy_config_files = Bool()
+    ipython_dir = Unicode()
+    log_datefmt = Unicode()
+    log_format = Unicode()
 
     # The classes added here determine how configuration will be documented
     classes = List()
 
+    # Basic configuration instance
+    _basic_config = Instance(BasicConfig)
+
     def _classes_default(self):
-        return [ProfileDir]
+        return [BasicConfig]
+
+    def _config_file_name_default(self):
+        return u'nbgrader_config.py'
```

Comment:
I have to check, but you "might" be able to drop the `.py` now that we have also json files.
Not sure.